synapse/config/workers.py

```diff
@@ -296,6 +304,94 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
             self.worker_name is None and background_tasks_instance == "master"
         ) or self.worker_name == background_tasks_instance
 
+    def _should_this_worker_perform_duty(
+        self,
+        config: Dict[str, Any],
+        legacy_master_option_name: str,
+        legacy_worker_app_name: str,
+        new_option_name: str,
+    ) -> bool:
+        """
+        Figures out whether this worker should perform a certain duty.
+
+        This function is temporary and is only to deal with the complexity
+        of allowing old, transitional and new configurations all at once.
+
+        Contradictions between the legacy and new part of a transitional configuration
+        will lead to a ConfigError.
+
+        Parameters:
+            config: The config dictionary
+            legacy_master_option_name: The name of a legacy option, whose value is boolean,
+                specifying whether it's the master that should handle a certain duty.
+            legacy_worker_app_name: The name of a legacy Synapse worker application name
+                that would traditionally perform this duty.
```

Comment:
`The name of a legacy Synapse worker application name`
wording nit: we've repeated `name` twice.

Could we also add an example: "synapse.app.appservice" ?